src/platform/monitoring/DowntimeNotification/util/dismissedDowntimeNotifications.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Simple data structure that abstracts away add/removing dismissed flags in the session.
+ * The Downtime Approaching warning should only be shown once, so we store that flag in the session.
+ * We store it using the appTitle (which should be unique to the app) in an array so that other apps
+ * that may be experiencing downtime will still have the warning.
+ */
+const dismissedDowntimeNotifications = {
+  key: 'downtime-notifications-dismissed',
+  setup() {
+    const rawData = window.sessionStorage[this.key];
+    this._dismissedDowntimeNotifications = rawData ? JSON.parse(rawData) : [];
+  },
+  contains(appTitle) {
+    return this._dismissedDowntimeNotifications.find(_appTitle => _appTitle === appTitle);
+  },
+  push(appTitle) {
+    this._dismissedDowntimeNotifications.push(appTitle);
+    window.sessionStorage[this.key] = JSON.stringify(this._dismissedDowntimeNotifications);
+  }
+};
+
```

Comment:
I really don't like this code but it was a fine way to abstract away all of the tedious `sessionStorage` stuff.

Reply:
Could using a set for `this._dismissedDowntimeNotifications` be of interest?

Reply:
Yeah, I think a Set would make sense here.

Reply:
Also, I think this is pretty clear use case for Redux: this is application state and it's shared among different components.

Reply:
I suppose that I could move the handler for closing the DowntimeApproaching modal into an action and have that also drive this data.